packages/astro/components/ViewTransitions.astro

```diff
@@ -65,19 +65,19 @@ const { fallback = 'animate' } = Astro.props as Props;
 	};
 
 	async function getHTML(href: string) {
-		let res;
 		try {
-			res = await fetch(href);
+			const res = await fetch(href);
+			const html = await res.text();
+			return {
+				ok: res.ok,
+				html,
+				redirected: res.redirected ? res.url : undefined,
+				// drop potential charset (+ other name/value pairs) as parser needs the mediaType
+				mediaType: res.headers.get('content-type')?.replace(/;.*/, ''),
```

Comment:
```suggestion
				mediaType: res.headers.get('content-type')?.replace(/;.*$/, ''),
```

Might just want to check that this replaces everything up to the end of the value? In practice, probably doesn't make much of a difference but it seems slightly more correct to me.